ephios/plugins/questionnaires/models.py

```diff
@@ -0,0 +1,162 @@
+from django_select2.forms import Select2MultipleWidget, Select2Widget
+from rest_framework import serializers
+
+from ephios.core.models.events import AbstractParticipation, Shift
+from ephios.core.models.users import UserProfile
+from ephios.core.signup.participants import AbstractParticipant, LocalUserParticipant
+
+
+class Question(models.Model):
+    class Type(models.TextChoices):
+        TEXT = "text", _("Text input")
+        SINGLE = "single", _("Single choice")
+        MULTIPLE = "multiple", _("Multiple choice")
+
+    name = models.CharField(
+        max_length=50,
+        verbose_name=_("Name"),
+        help_text=_(
+            "The name of the question is used to identify the question, for example when picking questions for a shift"
+        ),
+    )
+    question_text = models.CharField(max_length=250, verbose_name=_("Question"))
+    description = models.TextField(verbose_name=_("Description"), blank=True, null=True)
+    required = models.BooleanField(
+        verbose_name=_("Required"),
+        help_text=_("Whether users must submit an answer to this question"),
+    )
+    type = models.CharField(
+        max_length=20, verbose_name=_("Type"), choices=Type.choices, default=Type.TEXT
+    )
+    choices = models.JSONField(default=list, verbose_name=_("Choices"))
+
+    class Meta:
+        verbose_name = _("Question")
+        verbose_name_plural = _("Questions")
+
+    def __str__(self):
+        return str(self.name)
+
+    def get_signup_form_field(
+        self, participant: AbstractParticipant, participation: AbstractParticipation, signup_choice
+    ):
+        # Restore answer from participation (when editing) or the user's saved answers (if local participant)
+        if existing_answer := Answer.objects.filter(
+            participation_id=participation.pk, question=self
+        ).first():
+            initial = existing_answer.answer
+        elif saved_answer := (
+            SavedAnswer.objects.filter(user=participant.user, question=self).first()
+            if isinstance(participant, LocalUserParticipant)
+            else None
+        ):
+            initial = saved_answer.answer
+        else:
+            initial = None
+
+        # Reset answer if selected option does not exist
+        if self.type in [self.Type.SINGLE, self.Type.MULTIPLE] and initial not in self.choices:
+            initial = None
+
+        field_name = self.get_form_slug()
+        required = self.required if initial is None else False
+
+        field = {
+            "form_class": None,
+            "form_kwargs": {},
+            "serializer_class": None,
+            "serializer_kwargs": {},
+        }
+
+        field["form_kwargs"] = {
+            "label": self.question_text,
+            "help_text": self.description,
+            "initial": initial,
+            "required": required,
+        }
+
+        field["serializer_kwargs"] = {
+            "required": required,
+        }
```

Comment:
can you put this into the dict literal in lines 70/72 please?

Reply:
Changed in 8e272dbcf39e850e08428f5a5bef47bddf18e723